packages/kmod-6.1-nvidia/nvidia-persistenced.service

```
@@ -0,0 +1,15 @@
+[Unit]
+Description=NVIDIA Persistence Daemon
+After=load-tesla-kernel-modules.service load-open-gpu-kernel-modules.service
+# The "nvidia" user must be created before this unit runs.
+Requires=systemd-sysusers.service
+After=systemd-sysusers.service
+
+[Service]
+Type=forking
+ExecStartPre=/bin/rm -rf /run/nvidia-persistenced
+ExecStart=/usr/libexec/nvidia/tesla/bin/nvidia-persistenced --user nvidia --verbose
+ExecStopPost=/bin/rm -rf /var/run/nvidia-persistenced
+
+[Install]
+RequiredBy=preconfigured.target
```

If this doesn't specifically need to run in an early phase of boot, I'd just put it with the Fabric Manager in `multi-user.target`:
```suggestion
[Install]
WantedBy=nvidia-persistenced.target
```

---

It can run at any time, but we should prefer that it runs earlier rather than later. Downstream customers may not always initialize the GPU device files themselves, so running this unit early ensures that those files are properly set by the time their units begin (as part of `multi-user.target` for example).

---

This should run before services like `ecs-gpu-init` so that we can rely on this service to create the devices.